### PR TITLE
added login and sign up page

### LINKED ISSUE #19: Log in and Sign up page
Use Mantine components to create Log in and Sign up forms with validation 

Figma design 

Image

### PR DESCRIPTION
**Pull request by SachiGoto**
### Feature Description

Sign up page and Log in page with form validation using Mantine 'use-form' 
Here are snapshots of how the forms look like. 
Screenshot 2023-07-02 at 3 09 10 AM

Screenshot 2023-07-02 at 3 32 20 AM

### Related Issue

https://github.com/van-squad/map-t3/issues/19#issue-1779513947 

### Something you want to mention

I installed npm install @mantine/form to use mantine use-form 
I think you need to do npm install to use it. 

Question 

As you can see I applied pretty identical css to log in and sign up forms. I'm wondering how I can avoid repeating the same css code for those forms. Any suggestions? 

### Checklist before requesting a review

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings

### CODE REVIEW

**Review by devjoylee — CHANGES_REQUESTED** [review]
Thank you Sachi for creating form components.

I noticed that the `@mantine/form` is missing from the `package.json` file in this PR. You didn't push it together I guess? It causes a build error, and the deployment has failed. 
After making a PR, please check the deploy preview section to make sure it has been successfully built. If it's failed, you can review their log and see what happened.

스크린샷 2023-07-04 오전 11 26 09

스크린샷 2023-07-04 오전 11 29 39

<br/>
<br/>

Just wanted to let you know that you can also connect this PR with the issue you made. You can add the related issue in the development section on the right side of the page. Then the issue will be automatically closed when this PR is merged.

250241586-30215a53-04bc-4868-9b4c-44b3258d2ff1

**Comment by SachiGoto** [comment]
Hi Joy, thank you for pointing out the building error! I didn't push package.json so I just pushed it but looks like deployment still failed at building stage. I will look into it! 

Also, I didn't know that I can link the PR with the issue! That's cool. I linked tasks with the corresponding issues tho.

**Comment by Lada496** [comment]
> Hi Joy, thank you for pointing out the building error! I didn't push package.json so I just pushed it but looks like deployment still failed at building stage. I will look into it!
> 
> Also, I didn't know that I can link the PR with the issue! That's cool. I linked tasks with the corresponding issues tho.

According to the deploy error, I assume you're using new Button in you're local while this repo doesn't have it. 
You can cherry-pick #23 commits or wait until it's merged into `develop` and then merge `develop` into this branch.

**Comment by Lada496** [comment]
I think there are multiple ways to avoid having duplicated css.
If it's s big project, we can have Form components including TextInput, Checkbox, ...etc to share them everywhere in the app. For small apps like this, I think using [Route Groups](https://nextjs.org/docs/app/building-your-application/routing/colocation#route-groups) is enough. In this structure, we can have sharing stylings for both `login` and `signup`.
I've seen some code that combining both login and singup in one component but personally I don't like it because we have to use conditional logic everywhere and dealing with two different things in one place is not an ideal way.
```
// image
├── app
│   ├── (auth)
│   |    │   ├── styles.ts // and anything you think sharable (maybe frame component for example)
│   │    │   ├── login // go to /login
│   │    │   │   ├── page.tsx
│   │    │   │    │   ├── components ...
│   │    │   ├── signup  // go to /signup
│   │    │   │   ├── page.tsx
│   │    │   │    │   ├── components ...
```

**Review by devjoylee — APPROVED** [review]

**Comment by SachiGoto** [comment]
That sounds good! I initially made a css module and added css styles there but I wasn't able to use theme color scheme to set css for light mode and dark mode like how I can do with using createStyles. 

So if it's a ts file, I can use createStyles.. ?

File: src/clients/pages/Signup/SignupForm/index.tsx
Comment (Lada496) [inline src/clients/pages/Signup/SignupForm/index.tsx]: @SachiGoto I meant to separate `useStyle()` and each component into different files and export/import `useStyle()`. Then we can share the same stylings with login/signup

Comment (SachiGoto) [reply src/clients/pages/Signup/SignupForm/index.tsx]:
```
styles.ts 


import {
  createStyles,
  rem,
} from "@mantine/core";

export const formStyles = createStyles((theme) => ({
  frame1: {
    height: "75vh",
    width: "50vw",
    maxWidth: "35rem",
    borderRadius: "2.5rem",
    position: "relative",
    border: `1px solid ${
      theme.colorScheme === "dark" ? theme.colors.blue[1] : "black"
    }`,
    [theme.fn.smallerThan("sm")]: {
      width: "80vw",
    },
  },
  frame2: {
    width: "100%",
    maxWidth: "35rem",
    display: "flex",
    flexDirection: "column",
    borderRadius: "2.5rem",
    border: `1px solid ${
      theme.colorScheme === "dark" ? theme.colors.blue[1] : "black"
    }`,
    position: "absolute",
    top: "8px",
    right: "9px",
    height: "100%",
    [theme.fn.smallerThan("sm")]: {
      width: "80vw",
    },
  },
  title: {
    color: theme.colorScheme === "dark" ? theme.white : theme.black,
  },

  form: {
    paddingTop: rem(80),
    backgroundColor: "transparent",
    width: "80%",
    display: "flex",
    flexDirection: "column",
    alignItems: "start",
    justifyContent: "center",
    margin: "3% auto",

    [theme.fn.smallerThan("sm")]: {
      maxWidth: "100%",
    },
  },
}));




```

```
LoginForm.tsx 

import {formStyles} from "~/clients/pages/style";
<div className={formStyles.frame1}></div>
```

So I created style.ts and moved createStyle css there and called it formStyles. 
I was trying and see if I can import it and use some of the styles but I got an error... 
What am I doing wrong... ?? 

Screenshot 2023-07-07 at 5 11 18 PM

Comment (Lada496) [reply src/clients/pages/Signup/SignupForm/index.tsx]: it seems like you forgot to put `const {classes} = formStyles()`
And others should be remain the same.

Comment (SachiGoto) [reply src/clients/pages/Signup/SignupForm/index.tsx]: Ahhhhhh you are so right.  Now it works! Thank you so much Yuko! Also, how come its ts file not tsx file?

Comment (Lada496) [reply src/clients/pages/Signup/SignupForm/index.tsx]: because it's just a function, not a react component

**Review by Lada496 — APPROVED** [review]

**Pull request merged**
